src/features/auth/login/PasswordForm.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { useForm } from "react-hook-form";
+import { Redirect, useHistory, useLocation } from "react-router-dom";
+import { loginRoute } from "../../../AppRoutes";
+import Button from "../../../components/Button";
+import TextInput from "../../../components/TextInput";
+import { useAppDispatch, useTypedSelector } from "../../../store";
+import { passwordLogin } from "../authActions";
+import { clearError } from "../authSlice";
+
+export default function PasswordForm() {
+  const dispatch = useAppDispatch();
+  const authLoading = useTypedSelector((state) => state.auth.loading);
+
+  const { handleSubmit, register } = useForm<{ password?: string }>();
+  const history = useHistory();
+  const location = useLocation<{ username?: string }>();
+
+  const onSubmit = handleSubmit(async (data: { password?: string }) => {
+    if (!data.password) return;
+    dispatch(
+      passwordLogin({
+        username: location.state.username!,
+        password: data.password,
+      })
+    );
+  });
+
+  const backClicked = () => {
+    dispatch(clearError());
+    history.push(loginRoute, location.state);
+  };
+
+  return (
+    <>
+      {!location.state?.username && <Redirect to={loginRoute} />}
```

Comment:
```suggestion
      {!location.state.username && <Redirect to={loginRoute} />}
```
According to the TS types, `state` is always present on the `location` object so no need to use optional chaining.

Reply:
I'll double check, but I think this line (or maybe somewhere else?) was causing an error because username didn't exist on the state object (the optional chain is for username, not the state). That's why I added all the option chains on the usernames, but I'll see which ones are really necessary.

(Actually I'm not sure why it would cause an error at all, I would have thought it just returns undefined which is falsey.)